frame/assets-freezer/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Assets Freezer Pallet
+//!
+//! An extension pallet for use with the Assets pallet for allowing funds to be locked and reserved.
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+/*
+pub mod weights;
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+pub mod mock;
+#[cfg(test)]
+mod tests;
+*/
+use sp_std::prelude::*;
+use sp_runtime::{
+	RuntimeDebug, TokenError, traits::{
+		AtLeast32BitUnsigned, Zero, StaticLookup, Saturating, CheckedSub, CheckedAdd,
+		StoredMapError,
+	}
+};
+use codec::{Encode, Decode, HasCompact};
+use frame_support::{ensure, dispatch::{DispatchError, DispatchResult}};
+use frame_support::traits::{Currency, ReservableCurrency, BalanceStatus::Reserved, StoredMap};
+use frame_support::traits::tokens::{WithdrawConsequence, DepositConsequence, fungibles, FrozenBalance};
+use frame_system::Config as SystemConfig;
+
+//pub use weights::WeightInfo;
+pub use pallet::*;
+use frame_benchmarking::frame_support::traits::fungibles::Inspect;
+
+type BalanceOf<T> = <<T as Config>::Assets as fungibles::Inspect<<T as SystemConfig>::AccountId>>::Balance;
+type AssetIdOf<T> = <<T as Config>::Assets as fungibles::Inspect<<T as SystemConfig>::AccountId>>::AssetId;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::DispatchResult,
+		pallet_prelude::*,
+	};
+	use frame_system::pallet_prelude::*;
+	use super::*;
+
+	/// The information concerning our freezing.
+	#[derive(Eq, PartialEq, Clone, Encode, Decode, RuntimeDebug, Default)]
+	pub struct FreezeData<Balance> {
+		/// The amount of funds that have been reserved. The actual amount of funds held in reserve
+		/// (and thus guaranteed of being unreserved) is this amount less `melted`.
+		///
+		/// If this `is_zero`, then the account may be deleted. If it is non-zero, then the assets
+		/// pallet will attempt to keep the account alive by retaining the `minimum_balance` *plus*
+		/// this number of funds in it.
+		pub(super) reserved: Balance,
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	/// The module configuration trait.
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The fungibles trait impl whose assets this reserves.
+		type Assets: fungibles::Inspect<Self::AccountId>;
+
+		/// Place to store the fast-access freeze data for the given asset/account.
+		type Store: StoredMap<(AssetIdOf<Self>, Self::AccountId), FreezeData<BalanceOf<Self>>>;
+
+//		/// Weight information for extrinsics in this pallet.
+//		type WeightInfo: WeightInfo;
```

Comment:
missing weights, not ready for review?

Reply:
Not yet; for some reason the bot saw fit to remove my `inprogress` label and introduce `pleasereview`.